hsds/basenode.py

```diff
@@ -61,7 +68,14 @@ def getHeadUrl(app):
     req_reg = head_url + "/register"
     log.info("register: {}".format(req_reg))
 
-    body = {"id": app["id"], "port": app["node_port"], "node_type": app["node_type"]}
+    log.warn("env:{}".format(os.environ))
+
+    if "is_dcos" in app:
+        outside_port = os.environ.get('PORT0')
```

Comment:
Rather than using os.environ here (note this would throw a KeyNotFound exception is PORT0 is not set) you can use config.get("PORT0").  Alternatively, you may want to put a dict of docs related stuff when you set "is_dcos". e.g. app["dcos_config"] = {'PORT0': 999}, and then refer to that here.

Reply:
fixed